xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/viewers/attachments.js

```diff
@@ -231,3 +173,73 @@ viewers.Attachments = Class.create({
 return XWiki;
 }(XWiki || {}));
 
+/**
+ * Delete attachments from AttachmentsTab.
+ */
+require(['jquery', 'xwiki-events-bridge'], function($) {
+  /**
+   * Getting the button that triggers the modal.
+   */
+  $(document).on('show.bs.modal', '#deleteAttachment', function(event) {
+      $(this).data('relatedTarget', $(event.relatedTarget));
+  });
+  /**
+   * Event on deleteAttachment button.
+   */
+  $(document).on('click', '#deleteAttachment input.btn-danger', function() {
+      var modal = $('#deleteAttachment');
+      var button = modal.data('relatedTarget');
+      var notification;
+      /**
+       * Ajax request made for deleting an attachment. Delete the HTML element on succes. Disable the delete button
+       * before the request is send, so the user cannot resend it in case it takes longer.
+       * Display error message on failure.
+       */
+      $.ajax({
+          url : button.prop('href'),
+          beforeSend : function() {
+          button.prop('disabled', true);
+          notification = new XWiki.widgets.Notification(
+              "$services.localization.render('core.viewers.attachments.delete.inProgress')", 'inprogress');
+          },
+          success : function() {
+          var attachment = button.closest('.attachment');
+          // Remove the corresponding HTML element from the UI and update the attachment count.
+          attachment.remove();
+          updateCount();
+          notification.replace(new XWiki.widgets.Notification(
+              "$services.localization.render('core.viewers.attachments.delete.done')", 'done'));
+          },
+          error: function() {
+          // The button is enabled in case of error.
+          button.prop('disabled', false);
+          notification.replace(new XWiki.widgets.Notification(
+              "$services.localization.render('core.viewers.attachments.delete.failed')", 'error'));
+          }
+      })
+  });
+  /**
+   * Updating the number of files in AttachmentsTab and in More actions menu.
+   */
+  var updateCount = function() {
+      var attachmentsTab = $('#Attachmentstab').find('.itemCount');
+      var attachmentsNumber = $("#Attachmentspane .attachment").size();
+      if(attachmentsTab) {
+          attachmentsTab.text("$services.localization.render('docextra.extranb', ['__number__'])".replace("__number__", attachmentsNumber));
+      };
+      if($('#tmAttachments').length) {
+          // All the sub-nodes of tmAttachments are added in a normalized form.
```

Comment:
This comment is not clear. We're calling ``normalize()`` because we want to modify the value of a text node and we want to be sure that all consecutive text nodes are merged.